### PR TITLE
Validate that entry is not live before updating crtian fields + post…

### PR DESCRIPTION
**Pull request by yossipapi**
… process update

### CODE REVIEW

File: api_v3/lib/types/entry/KalturaLiveEntry.php [hunk 1]
```diff
@@ -176,23 +175,57 @@ public function validateConversionProfile(entry $sourceObject = null)
 	 */
 	public function validateForUpdate($sourceObject, $propertiesToSkip = array())
 	{
-		/* @var $sourceObject LiveEntry */
-		if(isset($this->dvrStatus))
+		$stopedOnlyUpdateAttributes = array(
+				"dvrStatus" => array("validatePropertyChanged"), 
+				"dvrWindow" => array("validatePropertyChanged"), 
+				"recordStatus" => array("validatePropertyChanged","validateRecordedEntryId"), 
+				"conversionProfileId" => array("validatePropertyChanged","validateRecordedEntryId")
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/entry/KalturaLiveEntry.php]: What about recordingOptions?

File: api_v3/lib/types/entry/KalturaLiveEntry.php [hunk 2]
```diff
@@ -176,23 +175,57 @@ public function validateConversionProfile(entry $sourceObject = null)
...
 		}
 		
-		if(isset($this->dvrWindow))
+		parent::validateForUpdate($sourceObject, $propertiesToSkip);
+	}
+	
+	protected function validatePropertyChanged($sourceObject, $attr)
+	{
+		/* @var $sourceObject LiveEntry */
+		$getter = "get" . ucfirst($attr);
```
Comment (tan-tan-kanarek) [inline api_v3/lib/types/entry/KalturaLiveEntry.php]: Please use map-between-objects to get the right getter.

File: alpha/lib/model/LiveEntry.php [hunk 3]
```diff
@@ -121,7 +123,22 @@ public function generateFileName( $sub_type, $version = null)
...
 	
-	protected $decidingLiveProfile = false;
+	/**
+	 * Code to be run before updating the object in database
+	 * @param PropelPDO $con
+	 * @return bloolean
+	 */
+	public function preUpdate(PropelPDO $con = null)
+	{
+		if($this->isColumnModified(entryPeer::CONVERSION_PROFILE_ID))
```
Comment (tan-tan-kanarek) [inline alpha/lib/model/LiveEntry.php]: I think that if recording status or options changed, you also should clear the recorded and redirected ids.

File: api_v3/lib/types/entry/KalturaLiveEntry.php (hunk 1 above)
Comment (yossipapi) [inline api_v3/lib/types/entry/KalturaLiveEntry.php]: recordingOptions is insertOnly so no need to handle this use case.

Comment (tan-tan-kanarek) [inline api_v3/lib/types/entry/KalturaLiveEntry.php]: It can't be insert-only anymore, if user can add recording he might need to set the options as well.

Comment (yossipapi) [inline api_v3/lib/types/entry/KalturaLiveEntry.php]: You are right ... will change to support this as well.

**Comment by yossipapi** [comment]
@tan-tan-kanarek added support for clearing vod entry when recordingOptions has changed please view

**Pull request closed without merging**